core/src/main/scala/org/apache/spark/sql/delta/RowId.scala

```diff
@@ -27,6 +27,9 @@ import org.apache.spark.sql.SparkSession
  */
 object RowId {
 
+  val MISSING_HIGH_WATER_MARK: RowIdHighWaterMark =
```

Comment:
This provides a default high-water mark (-1), in many places it can be used transparently instead of manually handling the case where the high-water mark is not defined